kernel/syscall.c

```diff
@@ -2087,11 +2088,23 @@ long myst_syscall_execve(
     long ret = 0;
     const char** argv = NULL;
     myst_thread_t* current_thread = myst_thread_self();
+    myst_path_t* resolved_path = NULL;
 
-    /* ATTN: the filename should be resolved if not an absolute path */
-    if (!filename || filename[0] != '/')
+    if (!filename)
         ERAISE(-EINVAL);
 
+    // Resolve relative path
+    if (filename[0] != '/')
+    {
+        if (!(resolved_path = (myst_path_t*)malloc(sizeof(myst_path_t))))
+            ERAISE(-ENOMEM);
+
+        if (ret = myst_realpath(filename, resolved_path))
+            ERAISE(ret);
+
+        filename = resolved_path->buf;
```

Comment:
Suggest declare a new variable for this. Reusing a `const` parameter is not a good practice.

Reply:
Adding a new variable might be too confusing? How do you think about this way? No reusing `filename` and no new variable

```
argv[0] = resolved_path ? resolved_path->buf : filename;
```

Reply:
Would it be conflicted with the logic below? A simple way to just declare
`const char* path = filename;`
And use `path` throughout the function instead.